sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/LeaseRenewerImpl.java

```diff
@@ -34,66 +34,76 @@ public LeaseRenewerImpl(Lease lease, LeaseManager leaseManager, Duration leaseRe
-                logger.info(String.format("Partition %s: renewer task started.", self.lease.getLeaseToken()));
-                long remainingWork = this.leaseRenewInterval.toMillis() / 2;
+        logger.info(String.format("Partition %s: renewer task started.", self.lease.getLeaseToken()));
+        long remainingWork = this.leaseRenewInterval.toMillis();
+
+        try {
+            while (!cancellationToken.isCancellationRequested() && remainingWork > 0) {
+                Thread.sleep(100);
+                remainingWork -= 100;
+            }
+        } catch (InterruptedException ex) {
+            // exception caught
+            logger.info(String.format("Partition %s: renewer task stopped.", self.lease.getLeaseToken()));
+            return Mono.empty();
+        }
+
+        return Mono.just(self)
+            .flatMap(value -> self.renew(cancellationToken))
+            .repeat(() -> {
+                if (cancellationToken.isCancellationRequested()) return false;
+
+                long remainingWorkInLoop = this.leaseRenewInterval.toMillis();
 
                 try {
-                    while (!cancellationToken.isCancellationRequested() && remainingWork > 0) {
+                    while (!cancellationToken.isCancellationRequested() && remainingWorkInLoop > 0) {
                         Thread.sleep(100);
-                        remainingWork -= 100;
+                        remainingWorkInLoop -= 100;
                     }
                 } catch (InterruptedException ex) {
                     // exception caught
                     logger.info(String.format("Partition %s: renewer task stopped.", self.lease.getLeaseToken()));
+                    return false;
                 }
 
-                while (!cancellationToken.isCancellationRequested()) {
-                    self.renew().block();
-
-                    remainingWork = this.leaseRenewInterval.toMillis();
-
-                    try {
-                        while (!cancellationToken.isCancellationRequested() && remainingWork > 0) {
-                            Thread.sleep(100);
-                            remainingWork -= 100;
-                        }
-                    } catch (InterruptedException ex) {
-                        // exception caught
-                        logger.info(String.format("Partition %s: renewer task stopped.", self.lease.getLeaseToken()));
-                        break;
-                    }
-                }
-            } catch (RuntimeException ex) {
-                logger.error(String.format("Partition %s: renew lease loop failed.", self.lease.getLeaseToken()), ex);
-                self.resultException = ex;
-            }
-        });
+                return !cancellationToken.isCancellationRequested();
+            })
+            .then()
+            .onErrorResume(throwable -> {
+                logger.error(String.format("Partition %s: renew lease loop failed.", self.lease.getLeaseToken()), throwable);
```

Comment:
slf4j logger has support for place holder `{}`, you don't need String.format here. remove `String.format`
https://www.slf4j.org/manual.html

Reply:
Fixed